eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/Examples.hs

```diff
@@ -77,10 +78,10 @@ import Test.Cardano.Ledger.Core.KeyPair (KeyPair (..), mkAddr, mkWitnessesVKey)
 import Test.Cardano.Ledger.Core.Utils (mkDummySafeHash, testGlobals, unsafeBoundRational)
 import Test.Cardano.Ledger.Shelley.Arbitrary (RawSeed (..))
 
-data LedgerExamples era = LedgerExamples
+data LedgerExamples rule era = LedgerExamples
   { -- tx
     leTx :: Tx TopTx era
-  , leApplyTxError :: ApplyTxError era
+  , leApplyTxError :: NonEmpty (PredicateFailure (EraRule rule era))
```

Comment:
I believe it will make more sense to keep the `ApplyTxError` type family here, instead of parametrizing `LedgerExamples` on the `rule`, otherwise Consensus layer will incur a bunch of breakage and I don;t believe that it is even necessary to use a `PredicateFailure` type family here at all.
Could you please give it a try and let me know if you get stuck. I am more than sure that it should be possible.

Reply:
I was convinced this was going to work, but it doesn't because:
- having `ApplyTxError` here means that we need to use the concrete constructor `ShelleyApplyTxError` here:
    https://github.com/IntersectMBO/cardano-ledger/blob/6ad878b5135cdac997c610ad2dac615e7cee358f/eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/Examples.hs#L150-L151
- ..which fixes `era ~ ShelleyEra`
- ..which means we can't reuse this code in other eras, e.g. here in Allegra:
    https://github.com/IntersectMBO/cardano-ledger/blob/6ad878b5135cdac997c610ad2dac615e7cee358f/eras/allegra/impl/testlib/Test/Cardano/Ledger/Allegra/Examples.hs#L44-L50

Ideas?